src/main/java/com/ldtteam/structurize/placement/handlers/entity/IEntityHandler.java

```diff
@@ -0,0 +1,119 @@
+package com.ldtteam.structurize.placement.handlers.entity;
+
+import com.ldtteam.structurize.placement.handlers.placement.IPlacementHandler;
+import com.ldtteam.structurize.placement.structure.IStructureHandler;
+import net.minecraft.client.Minecraft;
+import net.minecraft.core.BlockPos;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.level.levelgen.structure.BoundingBox;
+import net.minecraft.world.phys.AABB;
+import net.minecraft.world.phys.HitResult;
+import net.minecraft.world.phys.Vec3;
+
+import java.util.List;
+
+/**
+ * Handler for entity placements.
+ */
+public interface IEntityHandler
+{
+    /**
+     * Checks if this handler can deal with this entity.
+     *
+     * @param entity the entity to check.
+     * @return       true if so; false if you need to keep looking.
+     */
+    boolean canHandle(final Entity entity);
+
+    /**
+     * Checks if it's permitted to place this entity.
+     * By default, if a handler is defined it's assumed to be permitted, otherwise denied.
+     *
+     * @param entity     the entity being placed.
+     * @param isCreative if placing in creative mode.
+     * @param isFancy    if placing in fancy (non-placeholder) mode.
+     * @return           true if permitted, false otherwise.
+     */
+    default boolean canPlace(final Entity entity, final boolean isCreative, final boolean isFancy)
+    {
+        return true;
+    }
+
+    /**
+     * Gets the adjusted position in the world for the given entity.
+     *
+     * @param entity the entity.
+     * @param pos    the anchor position in world.
+     * @return       the adjusted entity position.
+     */
+    default Vec3 adjustPosition(final Entity entity, final BlockPos pos)

```

Comment:
I would suggest using longer name than just pos, to emphasize that this is anchor or w/e it actually is
```suggestion
    default Vec3 adjustPosition(final Entity entity, final BlockPos worldRootPos)

```

Reply:
Fixed